pkg/cmd/server/origin/asset.go

```diff
@@ -164,7 +164,13 @@ func (c *AssetConfig) addHandlers(mux *http.ServeMux) error {
 	versions := sets.NewString()
 	versions.Insert(latest.Versions...)
 	versions.Insert(klatest.Versions...)
+	deadVersions := sets.NewString()
+	deadVersions.Insert(configapi.DeadKubernetesAPILevels...)
+	deadVersions.Insert(configapi.DeadOpenShiftAPILevels...)
 	for _, version := range versions.List() {
+		if deadVersions.Has(version) {
```

Comment:
need to filter version per-kind... our dead levels and k8s' might not be identical
```
deadOriginVersions := sets.NewString(configapi.DeadOpenShiftAPILevels...)
deadKubernetesVersions := sets.NewString(configapi.DeadKubernetesAPILevels...)
...
 			if latest.OriginKind(kind, version) {
 				if !deadOriginVersions.Has(version) {
 					originResources.Insert(resource)
 				}
 			} else {
 				if !deadKubernetesVersions.Has(version) {
 					k8sResources.Insert(resource)
 				}
 			}
```